cartographer/mapping_2d/sparse_pose_graph/optimization_problem.cc

```diff
@@ -257,16 +261,14 @@ void OptimizationProblem::Solve(const std::vector<Constraint>& constraints,
   }
   for (size_t trajectory_id = 0; trajectory_id != node_data_.size();
        ++trajectory_id) {
-    for (size_t node_data_index = 0;
-         node_data_index != node_data_[trajectory_id].size();
-         ++node_data_index) {
-      node_data_[trajectory_id][node_data_index].point_cloud_pose =
-          ToPose(C_nodes[trajectory_id][node_data_index]);
+    for (auto& index_node_data : node_data_[trajectory_id]) {
+      index_node_data.second.point_cloud_pose =
+          ToPose(C_nodes[trajectory_id][index_node_data.first]);
```

Comment:
Consistency: Use `.at(index_node_data.first)` which checks that the data we want to copy exists?